fluid/resnet50.py

```diff
@@ -240,59 +243,44 @@ def test(exe):
+        train_losses = []
         for batch_id, data in enumerate(train_reader()):
-            batch_start = time.time()
-            if iter == args.iterations:
+            if iters == args.skip_batch_num:
+                start_time = time.time()
+                num_samples = 0
+            if iters == args.iterations:
                 break
             if not args.use_fake_data:
                 image = np.array(map(lambda x: x[0].reshape(dshape),
                                      data)).astype('float32')
                 label = np.array(map(lambda x: x[1], data)).astype('int64')
                 label = label.reshape([-1, 1])
-            loss, acc, weight = exe.run(fluid.default_main_program(),
-                                feed={'data': image,
-                                      'label': label},
-                                fetch_list=[avg_cost, batch_acc, batch_size_tensor])
+            loss, acc, weight = exe.run(
+                fluid.default_main_program(),
+                feed={'data': image,
+                      'label': label},
+                fetch_list=[avg_cost, batch_acc, batch_size_tensor])
+            iters += 1
+            num_samples += len(label)
             accuracy.add(value=acc, weight=weight)
-            if iter >= args.skip_batch_num or pass_id != 0:
-                batch_duration = time.time() - batch_start
-                pass_duration += batch_duration
-                im_num += label.shape[0]
-            every_pass_loss.append(loss)
-            print("Pass: %d, Iter: %d, loss: %s, acc: %s" %
-                  (pass_id, iter, str(loss), str(acc)))
-            scalar0.add_record(total_iters, loss)
-            scalar1.add_record(total_iters, acc)
-            iter += 1
-            total_iters += 1
-
-        total_train_time += pass_duration
-        pass_train_acc = accuracy.eval()
-        pass_test_acc = test(exe)
-        print("Pass: %d, Loss: %f, Train Accuray: %f, Test Accuray: %f, Handle Images Duration: %f\n" %
-              (pass_id, np.mean(every_pass_loss), pass_train_acc, pass_test_acc, pass_duration))
-
-    if total_train_time > 0.0 and iter != args.skip_batch_num:
-        examples_per_sec = im_num / total_train_time
-        sec_per_batch = total_train_time / (iter * args.pass_num - args.skip_batch_num)
-
-        print('\nTotal examples: %d, total time: %.5f' % (im_num, total_train_time))
-        print('%.5f examples/sec, %.5f sec/batch \n' %
-              (examples_per_sec, sec_per_batch))
+            train_losses.append(loss)
+            train_accs.append(acc)
+            print("Pass: %d, Iter: %d, Loss: %f, Accuracy: %f" %
+                  (pass_id, iters, loss, acc))
+        # evaluation
+        if args.with_test:
+            pass_test_acc = test(exe)
+        print("Pass: %d, Loss: %f, Train Accuray: %f\n" %
+              (pass_id, np.mean(train_losses), np.mean(train_accs)))
+        train_elapsed = time.time() - start_time
```

Comment:
if args.with_test is True. the train_elapsed will include the test time。

Reply:
done.